heaps/min_heap.py

```diff
@@ -1,66 +1,112 @@
 class HeapNode:
   
-    def initialize(self, key, value):
+    def __init__(self, key, value):
         self.key = key
         self.value = value
+    
+    def __str__(self):
+        return str(self.value)
 
 
 class MinHeap:
 
     def __init__(self):
         self.store = []
+    
+    def __len__(self):
+        return len(self.store)
 
 
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(1)
```

Comment:
👍  However the space complexity is O(log n) due to `heap_up` function's recursive call stack.